python/pyarrow/parquet/core.py

```diff
@@ -1825,7 +1826,14 @@ def read_table(source, *, columns=None, use_threads=True,
         filesystem, path = _resolve_filesystem_and_path(source, filesystem)
         if filesystem is not None:
             source = filesystem.open_input_file(path)
-        # TODO test that source is not a directory or a list
+        if not (
+            isinstance(source, str)
```

Comment:
The `#TODO` pointed out to validate this was not a directory and checking whether this is a string is not validating whether source is a file name or a directory.